keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/ObjectMappers.kt

```diff
@@ -29,18 +29,20 @@ fun configuredObjectMapper(): ObjectMapper = ObjectMapper().configureForKeel()
  * Factory method for [YAMLMapper]s configured how we like 'em.
  */
 fun configuredYamlMapper(): YAMLMapper = YAMLMapper().configureForKeel().disable(USE_NATIVE_TYPE_ID)
-
-fun <T : ObjectMapper> T.configureForKeel(): T =
-  apply {
+fun <T : ObjectMapper> T.configureForKeel(): T {
+  val javaTimeModule = JavaTimeModule()
+  javaTimeModule.addSerializer(Instant::class.java,PrecisionSqlSerializer())
```

Comment:
Can we add a round trip test that uses the serializer and then deserializes to ensure precision is in fact kept? Other than that, this PR looks good

Reply:
It's actually tested indirectly via the ArtifactRepositoryTests file  but let me see about a direct test.  It'll only keep as much precision as it writes in the JSON fields.  So this WILL reduce precision on these columns from what the JVM could track - but it's never had that until java 17 it looks like so we're not losing anything - though I may try to do a "6" instead of "3" precision as docs say MySQL CAN handle that level.

Reply:
Added a test see what you think :)